### PR TITLE
Resolves Issue/PR templates

### PR DESCRIPTION
**Pull request by ve-varun-sharma**
## Pull Request Type

Dictionary of Definitions here:
https://ben.balter.com/2015/12/08/types-of-pull-requests/

- [x] Just a heads up 🙆
- [ ] Sanity check 🏃‍♂️
- [ ] Work in progress (WIP) 🏗️
- [ ] Early feedback ✍️
- [ ] Line-by-line review 📝
- [ ] Pull request to a pull request 🔀

## Description

The PR offers to add some additional documentation structure for github issues in the repo and PRs for smoother review process.
If this is unneeded or feels too much structure no worries at all! I thought this may help with improved clarity, faster resolution, and increase better collaboration via providing sufficient details!

When users go to the Github UI to create a Github Issue or Pull Request, the new templates in the .github folder will automatically pull in the templates ensuring that users follow the desired instructions in the templates.

This PR addresses this Github issue: https://github.com/DefangLabs/defang/issues/705

 

#### Caveats/Notes (optional)

This PR doesn't touch any direct code so this will not affect an code and thus features.

#### Prompts (optional)

Please feel free to iterate, add, edit, delete from the proposed templates.

### Screenshots (optional)

Not needed for this.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] Extended the README / documentation, if necessary
- [x] My changes generate no new warnings
- [x] Is the code readable? Prefer to see more self-explanatory and consistency
- [x] Is the code tested properly? i.e. where appropriate, we're testing well-explained failures, behaviours, fixes, etc

### CODE REVIEW

**Comment by lionello** [comment]
Thanks! Assigning to @jordanstephens for review, because he has done some work on this before.

**Comment by ve-varun-sharma** [comment]
> This is cool! Can we make it more concise though? Too much boilerplate makes it hard to find the meat of the issue/pr.
> 
> "This PR addresses this Github issue" doesn't trigger GitHub to link the issue to the PR, see https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword. Wonder if that could be part of the template?

Ah yes of course!  I can certainly utilize the key word triggering there.

on the template, definitely. Are there parts that you like/dislike? What would you ideally like to have in there? I can help make the edits :)

**Comment by ve-varun-sharma** [comment]
> This is cool! Can we make it more concise though? Too much boilerplate makes it hard to find the meat of the issue/pr.
> 
> "This PR addresses this Github issue" doesn't trigger GitHub to link the issue to the PR, see https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword. Wonder if that could be part of the template?

@lionello 
I implemented your feedback, cut out some sections, and added a section based on what you suggested! Let me know if there's anything else we want to update this with :)

Ready for review! 
tagging @jordanstephens  for visibility here too

**Review by jordanstephens — APPROVED** [review]
Thanks for submitting these templates. I appreciate the spirit of these changes, though I think these templates are a bit heavy for our current stage. I've made some suggestions inline to tighten them up for our use-case.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- Operating system (e.g., Windows, macOS, Linux).
+- Browser (if applicable).
+- Version of Defang or relevant software.
+- Any relevant dependencies.
+
+## Screenshots/Logs:
+
+- Attach any relevant screenshots, error messages, or log files.
+
+## Additional Information:
```
[the remark this answers is not in the document]
Comment (ve-varun-sharma) [reply .github/ISSUE_TEMPLATE.md:25]: agreed!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+## Checklist before requesting a review
+
+- [ ] I have performed a self-review of my code
+- [ ] If it is a core feature, I have added thorough tests.
+- [ ] Extended the README / documentation, if necessary
+- [ ] My changes generate no new warnings
+- [ ] Is the code readable? Prefer to see more self-explanatory and consistency
+- [ ] Is the code tested properly? i.e. where appropriate, we're testing well-explained failures, behaviours, fixes, etc
+
+##### Memes Are welcomed
```
[the remark this answers is not in the document]
Comment (ve-varun-sharma) [reply .github/PULL_REQUEST_TEMPLATE.md:52]: Good callout here! I agree with your line of thinking. I like the simpler approach for this too

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## Expected behavior.
+
+## Actual behavior.
+
+## Environment:
+
+- Operating system (e.g., Windows, macOS, Linux).
+- Browser (if applicable).
+- Version of Defang or relevant software.
+- Any relevant dependencies.
```
[the remark this answers is not in the document]
Comment (ve-varun-sharma) [reply .github/ISSUE_TEMPLATE.md:19]: Nice changes to this too!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (lionello) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+## Linked Issues
+
+<!-- See https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue -->
+
+## Checklist
+
+- [ ] I have performed a self-review of my code
+- [ ] I have added appropriate tests
+- [ ] I have updated the Defang CLI help-text to reflect my changes, if necessary
```
Comment (lionello) [inline .github/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion
- [ ] I have updated the Defang CLI docs and/or README to reflect my changes, if necessary
```

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Environment:
+
+- Defang Version (`defang version`):
+- Operating system: (e.g., Windows, macOS, Linux)
+- Shell (or Browser, if applicable):
+
+## Additional Information, Relevant dependencies, Screenshots, Logs, Etc...
+
+Any other relevant details that might be helpful.
```
Comment (lionello) [inline .github/ISSUE_TEMPLATE.md:20]:
```suggestion
<!--Any other relevant details that might be helpful. -->
```

**Review by lionello — APPROVED** [review]
LGTM

**Pull request merged**